paimon-core/src/main/java/org/apache/paimon/KeyValueFileStore.java

```diff
@@ -238,7 +238,8 @@ private KeyValueFileStoreScan newScan(boolean forWrite) {
                 options.scanManifestParallelism(),
                 options.deletionVectorsEnabled(),
                 options.mergeEngine(),
-                options.changelogProducer());
+                options.changelogProducer(),
+                options.fileIndexReadEnabled() && options.deletionVectorsEnabled());
```

Comment:
Add "File index support Append Only tables and pk table with DV" in [fileIndex doc](https://paimon.apache.org/docs/master/concepts/spec/fileindex/)